File: problems/3090-maximum-length-substring-with-two-occurrences/solution.c
int maximumLengthSubstring(char* s) {
    int maxLen = 0;
    int n = 0;
    while (s[n] != '\0') {
        n++;
    }

    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            int counts[26] = {0};
            int valid = 1;
            for (int k = i; k <= j; k++) {
                counts[s[k] - 'a']++;
            }

            for (int k = 0; k < 26; k++) {
                if (counts[k] > 2) {
                    valid = 0;
                    break;
                }
            }

            if (valid) {
                int currentLen = j - i + 1;
                if (currentLen > maxLen) {
                    maxLen = currentLen;
                }
            }
        }
    }

    return maxLen;
}

File: problems/3090-maximum-length-substring-with-two-occurrences/solution.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        n = len(s)
        max_len = 0

        for i in range(n):
            for j in range(i, n):
                sub = s[i:j+1]
                counts = {}
                valid = True
                for char in sub:
                    counts[char] = counts.get(char, 0) + 1
                    if counts[char] > 2:
                        valid = False
                        break

                if valid:
                    max_len = max(max_len, len(sub))

        return max_len